Client/Frontend/Login Management/LoginListViewController.swift

```diff
@@ -11,19 +11,54 @@ import Shared
 private struct LoginListUX {
     static let RowHeight: CGFloat = 58
     static let SearchHeight: CGFloat = 58
+    static let selectionButtonFont = UIFont.systemFontOfSize(16)
+    static let selectionButtonTextColor = UIColor.whiteColor()
+    static let selectionButtonBackground = UIConstants.HighlightBlue
+}
+
+private extension UITableView {
+    var allIndexPaths: [NSIndexPath] {
+        return (0..<self.numberOfSections).flatMap { sectionNum in
+            (0..<self.numberOfRowsInSection(sectionNum)).map { NSIndexPath(forRow: $0, inSection: sectionNum) }
+        }
+    }
 }
 
 private let LoginCellIdentifier = "LoginCell"
 
 class LoginListViewController: UIViewController {
 
-    private var loginDataSource: LoginCursorDataSource = LoginCursorDataSource()
-    private var loginSearchController: LoginSearchController? = nil
+    private lazy var loginSelectionController: ListSelectionController = {
+        return ListSelectionController(tableView: self.tableView)
+    }()
+
+    private lazy var loginDataSource: LoginCursorDataSource = {
+        return LoginCursorDataSource()
+    }()
 
     private let profile: Profile
 
     private let searchView = SearchInputView()
 
+    private var activeLoginQuery: Success?
```

Comment:
I'm a little concerned that this is prone to races, but I think the impact of races won't be bad, so we're OK.